packages/loader/container-loader/src/container.ts

```diff
@@ -528,6 +531,20 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         this.removeAllListeners();
     }
 
+    public closeAndGetPendingLocalState(): string {
```

Comment:
From host perspective, if might be great to have a single API to save local state of container, that works both for attached container (this case) and detached (Draft mode, see Container.serialize()).
Then, on load, we could just picot based on some key saved in there to differentiate attached vs. non-attached cases.

Internally, they have to be two different workflows, of course...